onebusaway-android/src/amazon/java/org/onebusaway/android/map/googlemapsv2/StopOverlay.java

```diff
@@ -630,19 +630,20 @@ private void doFocusChange(ObaStop stop) {
         HashMap<String, ObaRoute> routes = mMarkerData.getCachedRoutes();
 
         // Notify listener
-        mOnFocusChangedListener.onFocusChanged(stop, routes);
+        mOnFocusChangedListener.onFocusChanged(stop, routes, stop.getLocation());
     }
 
     @Override
     public void onMapClick(LatLng latLng) {
         Log.d(TAG, "Map clicked");
 
-        // Only notify focus changed the first time the map is clicked away from a stop marker
         if (mMarkerData.getFocus() != null) {
             mMarkerData.removeFocus();
-            // Notify listener
-            mOnFocusChangedListener.onFocusChanged(null, null);
         }
+        // Set map clicked location
+        Location location = LocationUtil.makeLocation(latLng.latitude, latLng.longitude);
+        // Notify focus changed every time the map is clicked away from a stop marker
```

Comment:
@cagryInside The main change in logic here is that we are now notifying all listeners each time the user taps on the map, instead of only the first time the user taps on the map while a stop is selected.  From a quick look at BaseMapFragment and HomeActivity onFocusChanged() methods I don't think this will be a problem.  Main area at risk would be HomeActivity.onFocusedChanged(), since the initial assumption when I wrote that code was we'd only be moving the MyLocationButton down, hiding the ArrivalListFragment, etc. when a stop was currently selected.  Now, this code is executing each time you tap on the map, even when a stop isn't selected.  I think this should be ok, but please take a close look at it (including stepping through the debugger) to make sure nothing strange is happening there.  I'll take a look myself too.